### DISCUSSION TITLE
Validation react hook form

### DISCUSSION THREAD

**Discussion by kanu-shubham** [react-hook-form/react-hook-form#3679] [Q&A]
I have created a custom input. In that I have a button to validate. When I click on button I will make an api call to see if input is valid or not. Is there a way I can disable form submit  button or mark field as invalid?

**Comment by bluebill1049** [comment]
this new state is under request for comments: https://github.com/react-hook-form/react-hook-form/discussions/3669

**Reply by kanu-shubham** [reply]
Thanks for your reply .The one mentioned in 3669 is when I need to when the form is validating. In my case I want to know is there a way I can mark my form invalid or field invalid when api returns false. I cannot use validate as it gets called with onblur/ onchange and in my case I just want to validate field on button click

**Reply by bluebill1049** [reply]
codesandbox please

**Reply by kanu-shubham** [reply]
Ok. I will try to create a generic one. Cannot create same as like what I have.

**Reply by kanu-shubham** [reply]
One small thing does seterror disable submit button?
